auth/oidc.go

```diff
@@ -71,6 +75,20 @@ type OIDCClient struct {
 
 	// Verifier provides verification for ID Tokens.
 	Verifier *oidc.IDTokenVerifier
+
+	// A mutual exclusion lock for provider metadata.
+	mux sync.Mutex
```

Comment:
Typically name locks after the resource they affect, i.e.:

```suggestion
	providerLock sync.Mutex
```